core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -655,15 +655,19 @@ private[spark] object Utils extends Logging {
       // created the directories already, and that they are secured so that only the
       // user has access to them.
       getYarnLocalDirs(conf).split(",")
+    } else if (conf.getenv("SPARK_LOCAL_DIRS") != null) {
```

Comment:
What about the YARN options? Shall I leave that as the most preferable if running inside YARN?